### ISSUE TITLE
Using refs

### ISSUE THREAD

**Issue by denisnazarov** [mediachain/aleph#51]
How does one use refs to other objects?

Example:
https://github.com/MuseumofModernArt/collection has an Artists.json and Artworks.json. Artworks reference objects in Artists via an array of `ConstituentID`:

```json
{
  "Title": "Ferdinandsbrücke Project, Vienna, Austria, Elevation, preliminary version",
  "Artist": [
    "Otto Wagner"
  ],
  "ConstituentID": [
    6210
  ],
  "ArtistBio": [
    "Austrian, 1841–1918"
  ],
  "Nationality": [
    "Austrian"
  ],
  "BeginDate": [
    1841
  ],
  "EndDate": [
    1918
  ],
  "Gender": [
    "Male"
  ],
  "Date": "1896",
  "Medium": "Ink and cut-and-pasted painted pages on paper",
  "Dimensions": "19 1/8 x 66 1/2\" (48.6 x 168.9 cm)",
  "CreditLine": "Fractional and promised gift of Jo Carole and Ronald S. Lauder",
  "AccessionNumber": "885.1996",
  "Classification": "Architecture",
  "Department": "Architecture & Design",
  "DateAcquired": "1996-04-09",
  "Cataloged": "Y",
  "ObjectID": 2,
  "URL": "http://www.moma.org/collection/works/2",
  "ThumbnailURL": "http://www.moma.org/media/W1siZiIsIjU5NDA1Il0sWyJwIiwiY29udmVydCIsIi1yZXNpemUgMzAweDMwMFx1MDAzZSJdXQ.jpg?sha=137b8455b1ec6167",
  "Height (cm)": 48.6,
  "Width (cm)": 168.9
}
```

I want to include a reference to the Artist object via the statement's refs array:

- is this done via Artist WKI
- does it reference an object in another namespace explicitly (e.g. `museums.moma.artwork` and `museums.moma.artist`)?
- what if the reference is to a statement/object on a remote peer?

Other, probably not solved via refs:
- what if an object in `museums.moma.artist` represents the same person/entity in `museums.tate.artist`?
  - how can one mark them as the same thing
  - can i then query both museums for artworks if i know the WKI of an artist from one of them

**Comment by vyzo** [comment]
Two mechanisms supported at the lowest level:
- Multiple refs by wki can be attached to the same object in the statement body.
- Multiple objects can be referred to by the object itself, using the deps field in the statement body.

**Comment by parkan** [comment]
@denisnazarov the case you're describing goes like this:

* artist IDs get pulled out and converted into WKIs prefixed by that namespace, if known
* these WKIs are added to the deps field
* namespace-specific merge statement can be made equating two WKIs

Let me give you a simple example of merging based on ukiyo-e. For example, this image:

http://www.mfa.org/collections/object/226444

gets a WKI of `mfa:226444`, while this one:

http://collections.artsmia.org/art/2006/kirifuri-waterfall-at-mount-kurokami-in-shimotsuke-province-katsushika-hokusai

gets `mia:2006`. For simplicity, let's say that the prefixes and namespaces are the same. Then in the `ukiyoe` namespace, you have a statement w/o a body but with wki refs of `mfa:226444, mia:2006`, which says that from this perspective, these items constitute a single entity. Does that make sense?

**Comment by denisnazarov** [comment]
Yep makes total sense. I think the artist example is more exciting because it would let you find artworks in a different collection (picassos in tate) using an artist id in yours (picasso in moma).

Is this something we can illustrate with a query with what we have built today? If so would be great to get an example so I can include that in the tutorial/vid.

**Comment by parkan** [comment]
@denisnazarov yeah, I used this example for clarity, the artist case is equivalent. This kind of join is not supported yet, tho
